### DISCUSSION TITLE
How to use Hydro inside MVC view

### DISCUSSION THREAD

**Discussion by taublast** [hydrostack/hydro#37] [Q&A] [answered]
All examples are just for Razor pages while readme states this is MVC/Razor compatible.

If i include inside an MVC `Index.cshtml` page my component like that:

```
<div>     
    <hydro name="Counter" />
</div>
```

then in browser I get something like that and click is not working:

![image](https://github.com/hydrostack/hydro/assets/25801194/30146a87-1db1-42c8-b4d2-458e592e8214)

the counter is like in example:

```
@model  App.Pages.Components.Counter

<div>
    Count: <strong>@Model.Count</strong>
    <button hydro-on:click="@(() => Model.Add())">
        Add
    </button>
</div>
```

and 

```
using Hydro;

namespace App.Pages.Components
{
    public class Counter : HydroComponent
    {
        public int Count { get; set; }

        public void Add()
        {
            Count++;
        }
    }
}

```

**Comment by kjeske** [comment]
Hi! Have you added the tag helpers to your `_ViewImports.cshtml` file, according to [the docs](https://usehydro.dev/introduction/getting-started.html#getting-started)?

```razor
@addTagHelper *, Hydro
@addTagHelper *, {Your project assembly name}
```

**Comment by taublast** [comment]
Yes, i did, in [Views]/_ViewImports.cshtml

```
@addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
@addTagHelper *, Hydro
@addTagHelper *, App


```

**Comment by taublast** [comment]
There must be something "awesome" with my project.. 
UPDATE:  I'm using controllersandviews and razorpages at the same. Created similar _ViewImports.cshtml for both folders and counter renders fine for MVC and razorpages but onclick is not attached.. Will investigate further.

**Reply by kjeske** [reply]
It looks like the tag helpers are not initialized for some reason, like `_ViewImports.cshtml` was ignored. Just to verify that, please add `@addTagHelper *, Hydro` on the top of your view `Counter.cshtml`.

**Reply by taublast** [reply]
![image](https://github.com/hydrostack/hydro/assets/25801194/e6f0bf03-3085-41c5-afa6-133ff587cd7d)

Razor. Looks like now it attached but clicking does nothing, `Add()` is not hit in debugger and nothing happens. Sorry for the mess..

**Reply by kjeske** [reply]
I don't see the content in the `head` tag, which should contain:

```razor
<meta name="hydro-config" />
<script defer src="~/hydro/hydro.js" asp-append-version="true"></script>
<script defer src="~/hydro/alpine.js" asp-append-version="true"></script>
```

**Reply by taublast** [reply, marked as answer]
It was there indeed. Nevertheless i made the razor version work by adding its appropriate layout and import inside `Pages`. Then i cannot explain why then the MVC version started to work as well. 😂 It remains a total mystery for me what exact change made it happen.

**Reply by kjeske** [reply]
Good to hear it started to work! Thanks for taking the time to try Hydro! 😄
